ksqldb-engine/src/main/java/io/confluent/ksql/util/SandboxedSharedKafkaStreamsRuntimeImpl.java

```diff
@@ -37,6 +40,17 @@ public SandboxedSharedKafkaStreamsRuntimeImpl(
         getSandboxStreamsProperties(sharedRuntime)
     );
 
+    Collection<NamedTopology> namedTopologies = collocatedQueries.values()
+        .stream()
+        .map(BinPackedPersistentQueryMetadataImpl::getTopology)
+        .collect(Collectors.toSet());
+
+    if (!namedTopologies.containsAll(sharedRuntime.kafkaStreams.getAllTopologies())
+        || !sharedRuntime.kafkaStreams.getAllTopologies().containsAll(namedTopologies)) {
+      log.warn("Streams topologies and registered queries do not align. \nmetadata: {} \nstreams: {}",
+          namedTopologies, sharedRuntime.kafkaStreams.getAllTopologies());
```

Comment:
Hmm...have you checked what all actually gets printed when you try to print a full `NamedTopology` object (ie in `#getAllTopologies`? Wondering if this is going to spam the logs with the full contents of the internal topology builder metadata, for example

If it's really long and ugly we may want to just print out the names only

Reply:
Good idea we will just print the names